### ISSUE TITLE
Verify all metrics are correctly logged to W&B

### ISSUE THREAD

**Issue by prahaladtalur** [prahaladtalur/N-CMAPSS-Engine-Prediction#7]
## Problem

Need to ensure all RUL prediction metrics are correctly logged to W&B for experiment tracking and comparison with published papers.

**Current gap:** Our CNN-GRU shows RMSE_norm=0.098, but we're not tracking this in W&B. Papers report RMSE_norm=0.021-0.032.

### Standard Metrics (Currently Tracked)
- [x] MSE, RMSE, MAE, MAPE (raw, in cycles)
- [x] R² Score
- [x] PHM Score (raw and normalized)
- [x] Asymmetric Loss
- [x] Accuracy@10, Accuracy@15, Accuracy@20

### New Metrics (from issue #6)
- [ ] **RMSE (normalized)** — RUL scaled to [0, 1]
- [ ] **MAE (normalized)** — RUL scaled to [0, 1]
- [ ] R² on normalized values (should match raw R²)

### Metadata to Track
- [ ] RUL normalization range (y_min, y_max)
- [ ] Model architecture details (layers, params)
- [ ] Training config (batch_size, lr, patience, etc.)
- [ ] Dataset info (FD1-FD7, sequence length)

## Current Status

**Baseline (CNN-GRU, asymmetric loss):**
```
Raw Metrics:
  RMSE:  6.37 cycles  ✓ tracked
  MAE:   4.84 cycles  ✓ tracked
  R²:    0.9128       ✓ tracked

Normalized Metrics:
  RMSE:  0.0980       ✗ NOT tracked
  MAE:   0.0745       ✗ NOT tracked
  R²:    0.9128       ✗ NOT tracked

Gap from SOTA:
  RMSE:  3.1× worse   ✗ NOT visible in W&B
  MAE:   2.9× worse   ✗ NOT visible in W&B
```

### 1. Audit Current W&B Logging

Check `train_model.py` for:
- [ ] WandbCallback configuration
- [ ] Test metrics logging (not just val)
- [ ] Metric naming conventions
- [ ] Summary vs history logging

### 2. Add Normalized Metrics Logging

```python
# After final test evaluation
test_metrics = compute_all_metrics(y_test_true, y_test_pred)

wandb.log({
    # Raw metrics (existing)
    "test/rmse": test_metrics["rmse"],
    "test/mae": test_metrics["mae"],
    "test/r2": test_metrics["r2"],
    "test/phm_score": test_metrics["phm_score_normalized"],
    "test/acc_10": test_metrics["accuracy_10"],
    "test/acc_20": test_metrics["accuracy_20"],
    
    # NEW: Normalized metrics (from issue #6)
    "test/rmse_normalized": test_metrics["rmse_normalized"],
    "test/mae_normalized": test_metrics["mae_normalized"],
    
    # NEW: Gap from SOTA
    "test/rmse_norm_gap": test_metrics["rmse_normalized"] / 0.032,  # vs paper best
    "test/mae_norm_gap": test_metrics["mae_normalized"] / 0.026,
})
```

### 3. Add Run Summary

```python
# Add to W&B summary (appears in dashboard)
wandb.summary.update({
    "best_rmse": test_metrics["rmse"],
    "best_rmse_normalized": test_metrics["rmse_normalized"],
    "best_r2": test_metrics["r2"],
    "model_architecture": "cnn_gru",
    "num_parameters": model.count_params(),
    "rul_range_min": y_min,
    "rul_range_max": y_max,
    "dataset": f"N-CMAPSS FD{fd}",
    "sequence_length": config["max_sequence_length"],
})
```

### 4. Add Custom W&B Charts

- [ ] Scatter plot: true RUL vs predicted RUL
- [ ] Error distribution histogram
- [ ] Normalized metrics comparison table
- [ ] Performance gap tracker (vs SOTA)

### 5. Verify Metric Calculations

- [ ] Cross-check `src/utils/metrics.py` implementations
- [ ] Ensure sklearn metrics match our custom ones
- [ ] Add unit tests for metric functions
- [ ] Validate normalized RMSE calculation

### 6. Update Documentation

- [ ] Document each metric in README
- [ ] Add interpretation guide (what's "good"?)
- [ ] Explain raw vs normalized metrics
- [ ] Link to paper benchmarks

## Example W&B Run Summary

```yaml
Model: CNN-GRU (asymmetric loss)
Dataset: N-CMAPSS FD1
Sequence Length: 1000

Raw Metrics:
  RMSE: 6.37 cycles
  MAE:  4.84 cycles
  R²:   0.9128
  Acc@20: 99.12%

Normalized Metrics:
  RMSE: 0.0980  (target: <0.032, gap: 3.1×)
  MAE:  0.0745  (target: <0.026, gap: 2.9×)
  R²:   0.9128  (target: >0.987, gap: 0.07)

Status: 🟡 Needs Improvement (MDFA architecture, issue #8)
```

## Acceptance Criteria
- [ ] All metrics from `compute_all_metrics()` logged to W&B
- [ ] Test metrics logged with `test/` prefix
- [ ] Run summary includes normalized metrics
- [ ] Gap from SOTA visible in dashboard
- [ ] Metric calculations verified with unit tests
- [ ] W&B dashboard shows normalized RMSE chart
- [ ] Easy to compare runs: CNN-GRU vs MDFA vs Transformer

## Related Issues
- Issue #6: Add normalized RMSE/MAE calculation
- Issue #8: MDFA implementation to close 3× gap

## Priority
**High** — blocking proper comparison with published results and tracking progress toward SOTA.

**Comment by prahaladtalur** [comment]
All W&B metrics are fully implemented and logged. See train_model.py lines 322-354 and README.md metrics documentation. 

**Metrics currently logged:**
- ✅ RMSE (normalized) 
- ✅ MAE (normalized)
- ✅ R² Score
- ✅ PHM Score (normalized)
- ✅ Accuracy@10, Accuracy@15, Accuracy@20
- ✅ SOTA gap calculations (rmse_normalized_gap, mae_normalized_gap)
- ✅ Comprehensive visualizations (training history, predictions, error distribution)
- ✅ Summary metadata with model/data/training info

The README.md has been updated with comprehensive documentation of all metrics (commit 4cd71b4).
